contact-form/tests/send-email.test.js

```diff
@@ -0,0 +1,133 @@
+      expect(result._statusCode).toEqual(200);
+      expect(result._body.success).toBe(true);
+      expect(setApiKeyMock).toHaveBeenCalled();
+      done();
+    };
+
+    sendMock.mockResolvedValue(() => ({}));
+
+    const event = {
+      from: 'from2@example.com',
+      content: 'Hello',
+    };
+    sendEmail(context, event, callback);
+  });
+
+  it('returns a 400 response when from is blank', (done) => {
+    const callback = (_err, result) => {
+      expect(result).toBeInstanceOf(Twilio.Response);
+      expect(result._statusCode).toEqual(400);
+      expect(result._body.success).toBe(false);
+      expect(result._body.error).toEqual("Please set a 'from' email.");
+      expect(sendMock).not.toHaveBeenCalled();
+      expect(setApiKeyMock).not.toHaveBeenCalled();
+      done();
+    };
+
+    const event = {
+      from: '',
+      content: 'Hello',
+    };
+    sendEmail(context, event, callback);
+  });
+
+  it('returns a 400 response when content is blank', (done) => {
+    const callback = (_err, result) => {
+      expect(result).toBeInstanceOf(Twilio.Response);
+      expect(result._statusCode).toEqual(400);
+      expect(result._body.success).toBe(false);
+      expect(result._body.error).toEqual(
+        'Please enter some content for the email.'
+      );
+      expect(sendMock).not.toHaveBeenCalled();
+      expect(setApiKeyMock).not.toHaveBeenCalled();
+      done();
+    };
+
+    const event = {
+      from: 'from2@example.com',
+      content: '',
+    };
+    sendEmail(context, event, callback);
+  });
+
+  it('returns a 400 response when it fails to send an email', (done) => {
+    const callback = (_err, result) => {
+      expect(result).toBeInstanceOf(Twilio.Response);
+      expect(result._statusCode).toEqual(400);
+      expect(result._body.success).toBe(false);
+      expect(result._body.error).toEqual('Could not send email');
+      expect(setApiKeyMock).toHaveBeenCalled();
+      done();
+    };
+
+    sendMock.mockRejectedValue({
+      message: 'Could not send email',
+    });
+
+    const event = {
+      from: 'from2@example.com',
+      content: 'Hello',
+    };
+    sendEmail(context, event, callback);
+  });
+
+  it('returns a 400 response when it fails to send an email and returns the API error', (done) => {
+    const callback = (_err, result) => {
+      expect(result).toBeInstanceOf(Twilio.Response);
+      expect(result._statusCode).toEqual(400);
+      expect(result._body.success).toBe(false);
+      expect(result._body.error).toEqual('Something went wrong');
+      expect(setApiKeyMock).toHaveBeenCalled();
+      done();
+    };
+
+    sendMock.mockRejectedValue({
+      message: 'Could not send email',
+      response: {
+        body: {
+          errors: ['Something went wrong'],
+        },
+      },
+    });
+
+    const event = {
+      from: 'from2@example.com',
+      content: 'Hello',
+    };
+    sendEmail(context, event, callback);
```

Comment:
you could add a test for the default value of the subject if not provided but not wildly important.

Reply:
This was a good call, because an undefined subject actually results in the subject `"[contactform] undefined"`. Practically not a problem, since an empty input will submit an empty string. But something I should have tested for!